test/irb/test_context.rb

```diff
@@ -656,6 +656,23 @@ def test_eval_input_with_long_exception
       $VERBOSE = verbose
     end
 
+    def test_prompt_main_escape
+      irb = IRB::Irb.new(IRB::WorkSpace.new("main\a\t\r\n"))
+      assert_equal("irb(main    )>", irb.prompt('irb(%m)>', nil, 1, 1))
+    end
+
+    def test_prompt_main_inspect_escape
+      main = Struct.new(:inspect).new("main\\n\nmain")
+      irb = IRB::Irb.new(IRB::WorkSpace.new(main))
+      assert_equal("irb(main\\n main)>", irb.prompt('irb(%M)>', nil, 1, 1))
+    end
+
+    def test_prompt_main_truncate
+      irb = IRB::Irb.new(IRB::WorkSpace.new("a" * 100))
+      assert_match(/irb\(a.+\.\.\.\)>/, irb.prompt('irb(%m)>', nil, 1, 1))
```

Comment:
I'm not good with regexp, so I'd appreciate if we can still use `assert_equal` here to avoid regexp computation in my head 😂 

```rb
assert_equal(
  "irb(aaaaaaaaaaaaaaaaaaaaaaaaaaaaa...)>",
  irb.prompt('irb(%m)>', nil, 1, 1))
)
```